[ENH] Add GraphDB setup script

alyssadai opened the pull request:
<!--- Until this PR is ready for review, you can include [WIP] in the title, or create a draft PR. -->


<!---
Below is a suggested pull request template. Feel free to add more details you feel are relevant/necessary.

For more info on the PR process for Neurobagel repositories, see https://neurobagel.org/contributing/pull_requests/.
-->

<!-- 
Please indicate after the # which issue you're closing with this PR, if applicable.
If the PR closes multiple issues, include "closes" before each one is listed.
You can also link to other issues if necessary, e.g. "See also #1234".

https://help.github.com/articles/closing-issues-using-keywords
-->
- Closes #18 

<!-- 
Please give a brief overview of what has changed or been added in the PR.
This can include anything specific the maintainers should be looking for when they review the PR.
-->
Changes proposed in this pull request:

- Adds script that automates first-time GraphDB user + permissions setup, and new database setup
  - user setup steps can be turned off using a flag, if only a new database needs to be created
- Adds template `data-config.ttl` (for reviewer: maybe check if the location of the file makes sense?)

<!-- To be checked off by reviewers -->
## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

surchs commented on scripts/graphdb_setup.sh:
```diff
@@ -0,0 +1,176 @@
+curl -X PUT -u "admin:${ADMIN_PASS}" http://localhost:${NB_GRAPH_PORT_HOST}/${NB_GRAPH_DB} --data-binary "@data-config.ttl" -H "Content-Type: application/x-turtle"
+echo "Granting user ${NB_GRAPH_USERNAME} read/write permissions to database ${DB_NAME}..."
+curl -X PUT --header 'Content-Type: application/json' -d "
+{\"grantedAuthorities\": [\"WRITE_REPO_${DB_NAME}\",\"READ_REPO_${DB_NAME}\"]}" http://localhost:${NB_GRAPH_PORT_HOST}/rest/security/users/${NB_GRAPH_USERNAME} -u "admin:${ADMIN_PASS}"
```
so this is the thing I killed the openNeuro API with yesterday. What will happen here if I already have a database in the graph instance (i.e. I am only running this to add 1 new database)? 

Suppose
- `{DB_EXISTS_ALREADY}` is my existing database AND 
- `{NB_GRAPH_USERNAME}` has access to it AND
- `{DB_NAME}` is the new one I want to add
- Then I think this command will remove access to `{DB_EXISTS_ALREADY}`  for `{NB_GRAPH_USERNAME}` 

Could you check if that's true - and fix if yes

surchs replied:
Decision was: add warning with re-prompt for confirmation when user selects to just "make a new database". Purpose of warning is to remind user that this may remove existing permissions

surchs reviewed (approved):
Thanks @alyssadai this will be very useful! 

In light of recent lessons I learned the hard way, I have a question about DB access permissions. Please take a look before merging and address if needed

After that :cook:

alyssadai commented:
@surchs, sorry for the delay in addressing your review suggestions!

I've updated the script with the confirmation prompt for changing user permissions, as well as some more comments and a check before enabling security. Have a look and let me know if things make sense to you
